### ISSUE TITLE
Explicit SHMEM_BOUNCE_SIZE required for THREAD_MULTIPLE while using perf counter APIs

### ISSUE THREAD

**Issue by wrrobin** [Sandia-OpenSHMEM/SOS#709]
The newly added bounce buffer lock in the perf counter functions will lead to null pointer error for the default context in THREAD_MULTIPLE mode. Need to add check before acquiring lock.

**Comment by jdinan** [comment]
Are you sure?  Doesn't look like it to me.  We might want to add a "shmemx" unit test for these interfaces..

**Comment by wrrobin** [comment]
I think so. The following in `init` is causing this. Because the `BOUNCE_SIZE` is set to zero, there is no `free_list_init` for the bounce buffers in context. And, when we call the lock, we reference the bounce buffers to get the lock.
```
    if (shmem_internal_thread_level == SHMEM_THREAD_MULTIPLE &&
        !shmem_internal_params.BOUNCE_SIZE_provided) {
        shmem_internal_params.BOUNCE_SIZE = 0;
    }
```

**Comment by jdinan** [comment]
If the `SHMEMX_CTX_BOUNCE` option is not set on the context, the lock should get skipped.  Is it still being set?  I thought it should be cleared in the case you mentioned.  Might be good to make that unit test and see if you can get this failure to happen.

**Comment by wrrobin** [comment]
I will write the unit test and check this out. One use case already proved this scenario when I used the counter APIs from multi-threaded bandwidth tests. I traced the failure in that case and found that the lock is invoked but the bounce buffer is not initialized. After passing the explicit BOUNCE_SIZE, the issue got resolved.

**Comment by wrrobin** [comment]
Closed by #710
